core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -687,6 +692,46 @@ private RelationPlan createUpdatePlan(Analysis analysis, Update node)
         return new RelationPlan(commitNode, analysis.getScope(node), commitNode.getOutputSymbols(), Optional.empty());
     }
 
+    private RelationPlan createMergePlan(Analysis analysis, Merge node)
+    {
+        MergeNode mergeNode = new QueryPlanner(analysis, symbolAllocator, idAllocator, buildLambdaDeclarationToSymbolMap(analysis, symbolAllocator), metadata, Optional.empty(), session, ImmutableMap.of())
+                .plan(node);
+
+        TableFinishNode commitNode = new TableFinishNode(
+                idAllocator.getNextId(),
+                mergeNode,
+                mergeNode.getTarget(),
+                symbolAllocator.newSymbol("rows", BIGINT),
+                Optional.empty(),
+                Optional.empty());
+
+        return new RelationPlan(commitNode, analysis.getScope(node), commitNode.getOutputSymbols(), Optional.empty());
+    }
+
+    public static Optional<PartitioningScheme> createPartitioningScheme(Optional<NewTableLayout> writeTableLayout, List<Symbol> symbols, List<Integer> partitionFunctionArgumentIndexes)
```

Comment:
This method seems extracted from the method `createTableWriterPlan()`. Could you reuse the code?

Reply:
I tried, and it doesn't seem practical to do so.  The problem is the number of parameters, and especially the fact that to satisfy `createTableWriterPlan()`'s requirements, we'd have to add a new packaging class, because the method would need multiple return values.  Someday Java will Java will have multiple return values, and it will be easier to create commonly used APIs without needing garbage packaging classes for return values.